deploy/0_deploy.ts

```diff
@@ -41,82 +62,176 @@ const ensPublicResolver: PublicResolver = hre.artifacts.require(
-            if (contract == iexecPocoBoostDeployment.address) {
-                console.log(`[${i}] ${contract} (IexecPocoBoostDelegate) ${method}`);
-            }
-        }),
+    for (let i = 0; i < functionCount.toNumber(); i++) {
+        const [method, , contract] = await erc1538QueryInstance.functionByIndex(i);
+        console.log(`[${i}] ${contract} ${method}`);
+    }
+    const appRegistryAddress = await deploy(new AppRegistry__factory(), [], transferOwnershipCall);
+    const datasetRegistryAddress = await deploy(
+        new DatasetRegistry__factory(),
+        [],
+        transferOwnershipCall,
+    );
+    const workerpoolRegistryAddress = await deploy(
+        new WorkerpoolRegistry__factory(),
+        [],
+        transferOwnershipCall,
     );
+    // Set main configuration
+    const iexecAccessorsInstance = IexecAccessors__factory.connect(erc1538ProxyAddress, owner);
+    const iexecInitialized =
+        (await iexecAccessorsInstance.eip712domain_separator()) != ethers.constants.HashZero;
+    if (!iexecInitialized) {
+        await IexecMaintenanceDelegate__factory.connect(erc1538ProxyAddress, owner)
+            .configure(
+                rlcInstanceAddress,
+                'Staked RLC',
+                'SRLC',
+                9, // TODO: generic ?
+                appRegistryAddress,
+                datasetRegistryAddress,
+                workerpoolRegistryAddress,
+                ethers.constants.AddressZero,
+            )
+            .then((tx) => tx.wait());
+    }
+    // Set categories
+    const catCountBefore = await iexecAccessorsInstance.countCategory();
+    const categories = CONFIG.categories as Category[];
+    for (let i = catCountBefore.toNumber(); i < categories.length; i++) {
+        const category = categories[i];
+        await IexecCategoryManager__factory.connect(erc1538ProxyAddress, owner).createCategory(
+            category.name,
+            JSON.stringify(category.description),
+            category.workClockTimeRef,
+        );
+    }
+    const catCountAfter = await iexecAccessorsInstance.countCategory();
+    console.log(`countCategory is now: ${catCountAfter}`);
+    for (let i = 0; i < catCountAfter.toNumber(); i++) {
+        console.log(`Category ${i}: ${await iexecAccessorsInstance.viewCategory(i)}`);
+    }
 };
 
+/**
+ * Extract base contract name from contract factory name.
+ * Inputting `MyBoxContract__factory` returns `MyBoxContract`.
+ */
+function getBaseNameFromContractFactory(contractFactory: any) {
+    const name = contractFactory.constructor.name;
+    return name.replace('__factory', '');
+}
+
+/**
+ * Deploy through a GenericFactory a contract [and optionally trigger call]
+ */
+async function deploy(contractFactory: ContractFactory, constructorArgs?: any[], call?: string) {
+    let bytecode = constructorArgs
+        ? contractFactory.getDeployTransaction(...constructorArgs).data
+        : contractFactory.getDeployTransaction().data;
```

Comment:
```suggestion
    const args: any[] = constructorArgs ?? [];
    let bytecode = contractFactory.getDeployTransaction(...args).data
```
or shorter
```suggestion
    let bytecode = contractFactory.getDeployTransaction(...(constructorArgs ?? [])).data
```

Reply:
Better thanks! See [Get deployment bytecode with or without constructor args in single line](https://github.com/iExecBlockchainComputing/PoCo/pull/58/commits/1ae0b12e1e4d913659a13572d9fc61de5d8915e9)